### PR TITLE
#947 Overnight user activity emails

### PR DESCRIPTION
**Pull request by tbl73**
These are the emails that a user receives if activity has occurred on pages/collections they've worked in.  Note that I can't really write any new features tests for this work, because it's a rake task and isn't triggered that way.

### CODE REVIEW

File: app/views/user/update_profile.html.slim [hunk 1]
```diff
@@ -29,5 +29,28 @@
       tr
         th =f.label :location
         td =f.text_field :location, :placeholder => 'Where are you from?'
+
+    p.email_notifications Select which of the following notifications you want to receive:
+    table.form.email_notifications
```
Comment (saracarl) [inline app/views/user/update_profile.html.slim]: Where is the "Nightly emails on collections you work on" option?  (and let's list that first.)

File: app/views/user_mailer/nightly_user_activity.html.slim [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    p
+p
+-unless @active_note_pages.blank?
+  -@active_note_pages.each do |page|
+    =link_to "Click here", collection_transcribe_page_path(page.collection.owner, page.collection, page.work, page, only_path: false)
+    |&nbsp; to view the note on page #{page.title} in #{page.collection.title}: #{page.work.title}, which has been added since you worked on it.
+    p
+
+p Thanks, 
+|FromThePage
```
Comment (saracarl) [inline app/views/user_mailer/nightly_user_activity.html.slim]: We need a link to the settings where you adjust these emails at the end of all of the emails.

File: app/views/user/update_profile.html.slim [hunk 3]
```diff
@@ -29,5 +29,28 @@
...
+        td =f.check_box("notifications[add_as_owner]", {checked: @user.notification.add_as_owner})
+        th =f.label :add_as_owner, "Added to private work as an owner"
+      tr
+        td =f.check_box("notifications[note_added]", {checked: @user.notification.note_added})
+        th =f.label :page_edited, "Note added in response to a comment you made"
+      tr
+        td =f.check_box("notifications[work_added]", {checked: @user.notification.work_added})
+        th =f.label :work_added, "Work added to collection you've worked on"
+      tr
+        td =f.check_box("notifications[page_edited]", {checked: @user.notification.page_edited})
```
Comment (saracarl) [inline app/views/user/update_profile.html.slim]: This one needs to be unchecked by default.

Comment (tbl73) [reply app/views/user/update_profile.html.slim]: Which one needs to be unchecked by default?  There are a couple of different lines highlighted here.

Comment (saracarl) [reply app/views/user/update_profile.html.slim]: page edits.  I don't think you want all page edits emailed to you right away by default.

Comment (tbl73) [reply app/views/user/update_profile.html.slim]: Well, they aren't being immediately emailed under any circumstances - they're in the overnight emails.  Though you may still not want them sent to you by default; what do you think?

File: app/views/user/update_profile.html.slim (hunk 1 above)
Comment (tbl73) [reply app/views/user/update_profile.html.slim]: We don't have one of those, and we should - I think originally it was only page edits being sent overnight, and when we added additional information, I didn't think about it.

**Comment by tbl73** [comment]
We had some further discussion about the goals of this work, and this needs some changes - closing the PR and will re-open once it's ready.

**Pull request closed without merging**